pages/instructorQuestionManualGrading/instructorQuestionManualGrading.js

```diff
@@ -0,0 +1,128 @@
+        const params = {instance_question_id: res.locals.instance_question_id};
+
+        sqlDb.query(sql.instance_question_select_last_variant_with_submission, params, (err, result) => {
+            if (ERR(err, next)) return;
+            if (result.rowCount == 0) throw new Error('Instance question not found');
+    
+            const variant_id = result.rows[0].id;
+    
+            sqlDb.queryOneRow(sql.instance_question_select_question, params, (err, result) => {
+                if (ERR(err, next)) return;
+                if (result.rowCount == 0) throw new Error('Question not found');
+                res.locals.question = result.rows[0];
+
+                const params = [variant_id, null, true];
+
+                sqlDb.callZeroOrOneRow('variants_select_submission_for_grading', params, (err, result) => {
+                    if (ERR(err, next)) return;
+                    if (result.rowCount == 0) return new NoSubmissionError();
+                    const submission = result.rows[0];
+                    res.locals['submission'] = submission;
+                    
+                    sqlDb.callZeroOrOneRow('submissions_select', [submission_id], (err, result) => {
+                        if (ERR(err, next)) return;
+                        if (result.rowCount == 0) return new NoSubmissionError();
+                        const submission = result.rows[0];
+                        res.locals['submission'] = submission;
+                        res.locals['submission_updated'] = true;
+                        debug('_gradeVariantWithClient()', 'selected submission', 'submission.id:', submission.id);
+
+                        const params = [
+                            submission.id,
+                            res.locals.authn_user.user_id,
+                            submission.gradable,
+                            submission.broken,
+                            submission.format_errors,
+                            submission.partial_scores,
+                            score, // overwrite submission score
+                            submission.v2_score,
+                            {feedback:note}, // overwrite feedback
+                            submission.submitted_answer,
+                            submission.params,
+                            submission.true_answer,
+                        ];
+                    
+                        sqlDb.callOneRow('grading_jobs_insert', params, (err, result) => {
+                            if (ERR(err, next)) return;
+                    
+                            /* If the submission was marked invalid during grading the grading job will
+                               be marked ungradable and we should bail here to prevent LTI updates. */
+                            res.locals['grading_job'] = result.rows[0];
+                            if (!res.locals['grading_job'].gradable) return next(new NoSubmissionError());
```

Comment:
Use `error.make()` to create an error object, not `NoSubmissionError()`.

Reply:
That is definitely fine. Be aware that NoSubmissionError() is used in question.js, but I believe the nature of the error is slightly different.